plugins/modules/aws_s3.py

```diff
@@ -971,7 +971,8 @@ def main():
 
     if overwrite not in ['always', 'never', 'different', 'latest']:
         if module.boolean(overwrite):
-            overwrite = 'always'
+            # Fall to default configuration
+            overwrite = 'different'
```

Comment:
This would be overwrite=True, the module docs explicitly state that this is equivalent to 'always'.